src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -0,0 +1,343 @@
+        }
+
+        public void DecodeX509BasicConstraintsExtension(
+            byte[] encoded,
+            out bool certificateAuthority,
+            out bool hasPathLengthConstraint,
+            out int pathLengthConstraint)
+        {
+            throw new NotImplementedException();
+        }
+
+        public unsafe void DecodeX509BasicConstraints2Extension(
+            byte[] encoded,
+            out bool certificateAuthority,
+            out bool hasPathLengthConstraint,
+            out int pathLengthConstraint)
+        {
+            using (SafeBasicConstraintsHandle constraints = Interop.libcrypto.OpenSslD2I(Interop.libcrypto.d2i_BASIC_CONSTRAINTS, encoded))
+            {
+                Interop.libcrypto.BASIC_CONSTRAINTS* data = (Interop.libcrypto.BASIC_CONSTRAINTS*)constraints.DangerousGetHandle();
+                certificateAuthority = data->CA != 0;
+
+                if (data->pathlen != IntPtr.Zero)
+                {
+                    hasPathLengthConstraint = true;
+                    pathLengthConstraint = Interop.libcrypto.ASN1_INTEGER_get(data->pathlen).ToInt32();
+                }
+                else
+                {
+                    hasPathLengthConstraint = false;
+                    pathLengthConstraint = 0;
+                }
+            }
+        }
+
+        public byte[] EncodeX509EnhancedKeyUsageExtension(OidCollection usages)
+        {
+            throw new NotImplementedException();
+        }
+
+        public unsafe void DecodeX509EnhancedKeyUsageExtension(byte[] encoded, out OidCollection usages)
+        {
+            OidCollection oids = new OidCollection();
+
+            using (SafeEkuExtensionHandle eku = Interop.libcrypto.OpenSslD2I(Interop.libcrypto.d2i_EXTENDED_KEY_USAGE, encoded))
+            {
+                int count = Interop.NativeCrypto.GetX509EkuFieldCount(eku);
+
+                for (int i = 0; i < count; i++)
+                {
+                    IntPtr oidPtr = Interop.NativeCrypto.GetX509EkuField(eku, i);
+
+                    if (oidPtr == IntPtr.Zero)
+                    {
+                        throw new CryptographicException();
+                    }
+
+                    string oidValue = Interop.libcrypto.OBJ_obj2txt_helper(oidPtr);
+
+                    oids.Add(new Oid(oidValue));
+                }
+            }
+
+            usages = oids;
+        }
+
+        public byte[] EncodeX509SubjectKeyIdentifierExtension(byte[] subjectKeyIdentifier)
+        {
+            throw new NotImplementedException();
+        }
+
+        public unsafe void DecodeX509SubjectKeyIdentifierExtension(byte[] encoded, out byte[] subjectKeyIdentifier)
+        {
+            using (SafeAsn1OctetStringHandle octetString = Interop.libcrypto.OpenSslD2I(Interop.libcrypto.d2i_ASN1_OCTET_STRING, encoded))
+            {
+                subjectKeyIdentifier = Interop.NativeCrypto.GetAsn1StringBytes(octetString.DangerousGetHandle());
```

Comment:
Are we guaranteed here that octedString is a valid handle?  DangerousGetHandle is, well, "dangerous" and doesn't do any check.